core/state_transition.go

```diff
@@ -271,8 +271,12 @@ func (st *StateTransition) preCheck(gasBailout bool) error {
 			// libcommon.Hash{} means that the sender is not in the state.
 			// Historically there were transactions with 0 gas price and non-existing sender,
 			// so we have to allow that.
-			return fmt.Errorf("%w: address %v, codehash: %s", ErrSenderNoEOA,
-				st.msg.From().Hex(), codeHash)
+
+			// eip-7702 allows tx origination from accounts having delegated designation code.
+			if _, ok := st.state.GetDelegatedDesignation(st.msg.From()); !ok {
```

Comment:
Is this needed? I can see that `st.state.GetCodeHash()` does not return the   `codeHash` of the delegated code, even though `.GetCode()` returns the delegation string.

Reply:
GetCodeHash() does correspond to the delegation string; GetCode returns the delegation string.
Not sure if I understand the reasoning properly here.

A (EOA) delegates to B.
A makes a tx, its codeHash will be non-empty (and corresponds to delegation string). But since it's an EOA with delegated code, it should be allowed to make the tx.